src/Boo.Lang.Compiler/TypeSystem/Types.cs

```diff
@@ -121,27 +122,21 @@ public static class Types
 
 		public static readonly Type CompilerGeneratedAttribute = typeof(System.Runtime.CompilerServices.CompilerGeneratedAttribute);
 
-		static Type FindType(string typename, string typenamespace)
+		static Type FindType(string fullTypeName, string[] possibleAssemblyNames)
 		{
-			foreach (System.Reflection.Assembly assembly in AppDomain.CurrentDomain.GetAssemblies())
-			{
-				foreach (Type type in assembly.GetTypes())
```

Comment:
a bit more invasive, but also a bit more safe, as we actually wouldn't want to accidentally pick up a [ExtensionAttribute] from a user assembly.  very small risk, but I figured I'd do that while I was at it.